### DISCUSSION TITLE
insert adds 2-3 rows with EMPTY data

### DISCUSSION THREAD

**Discussion by sverp** [supabase/supabase#17897] [Questions] [answered]
1. Every time I load auth page I inserts 2-3 EMPTY rows and then row with data.
2. When loading/refreshing page sometimes the refresh token changes, shouldn't it be long lasting ?. The access token remained same.

This is the code for the page that is redirected to.

```
import { useEffect,useState } from "react"
import supabase from "../config/supabaseClient"

export default function AuthUser(){
    
    useEffect(()=> {
        async function getToken(){
            const { data, error } = await supabase.auth.getSession()
                if(data){
                    const { error } = await supabase
                    .from('credentials')
                    .insert({
                        user_id : data.session.user.id,
                        u_access_token : data.session.access_token,
                        u_refresh_token : data.session.refresh_token,
                    })
                    if(error){
                        console.log(error);
                    }
                }
                if(error){
                    console.log(error);
                }         
        }
        getToken();  
    },[])


    return (
        <>
            <h2>this is auth</h2>
        </>
    )
}

```

**Comment by GaryAustin1** [comment]
The refresh token gets used anytime the access_token expires or you do something that needs to refresh the session/access token.  Are you saying the expire time (exp claim) is not changing in the access token?

If you console.log your useEffect is it being called multiple times and is the data coming back null?  Null in data from getSession means the user is not in the process of signing in yet or will not be signed in.

You don't seem to check for getting anything in data when you go do the insert operation so if you got null back from getSession and did not set your state variables they would be blank.

**Reply by sverp** [reply]
1. I am getting access_token and refresh_token from getSession() to save it in the database and somehow refresh token is changing (which I think is wrong). Access token remains the same, I edited the code to what I am using in the question.I am using the same gmail account.
![tkon_prob](https://github.com/supabase/supabase/assets/37061618/69fb7dbe-f4a3-4b86-90eb-a0c7cabbbfc5)

2. You were right, now when adding check for data it only inserts one time.

**Reply by GaryAustin1** [reply, marked as answer]
The refresh token will change if used around 1 hour or more later than last refresh.  Every time the jwt expires (default 1hour) or is about to with auto refresh option on in client (default) a new refresh token will be returned and the old one "destroyed"  You can only use a refresh token 1 time to refresh the jwt.  But the refresh token does not expire it is good until it is used even if days later.
